src/fxcorrelator.py

```diff
@@ -201,20 +325,32 @@ def setup_sensors(self, katcp_server):
         """
         if not self._initialised:
             raise RuntimeError('Cannot set up sensors until instrument is initialised.')
+        ioloop = getattr(self, 'ioloop', None)
+        if not ioloop:
+            ioloop = getattr(katcp_server, 'ioloop', None)
+        if not ioloop:
+            # make my own ioloop
+            ioloop = IOLoop.instance()
+            ioloop.start()
+            ioloop.stop()  # move it to the end?
+
```

Comment:
I've added that bit to initiate ioloop and in line 317 used call_later and in 345 a add_callback